utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/extractor/JsonToTypedMap.scala

```diff
@@ -22,24 +23,22 @@ object JsonToTypedMap {
     def extract[A](fun: Json => Option[A], trans: A => AnyRef = identity[AnyRef] _): AnyRef =
       fun(json).map(trans).getOrElse(throw JsonToObjectError(json, definition))
 
-
     def extractObject(elementType: Map[PropertyName, SwaggerTyped], required: Set[PropertyName]): AnyRef = {
-      def nullOrError(jsonField: PropertyName): AnyRef = {
+      def noneOrError(jsonField: PropertyName): Option[AnyRef] = {
         if (required.contains(jsonField)) throw JsonToObjectError(json, definition)
-        else null
-      }
-
-      def notNullOrRequired(fieldName: String, value: Any, required: Set[PropertyName]): Boolean = {
-        value != null || required.contains(fieldName)
+        else None
       }
 
       extract[JsonObject](
         _.asObject,
         jo => TypedMap(
-          elementType.map {
-            case (jsonField, jsonDef) => jsonField -> jo(jsonField).map(JsonToTypedMap(_, jsonDef)).getOrElse(nullOrError(jsonField))
-          }
-            .filter(e => notNullOrRequired(e._1, e._2, required))
+          elementType
+            .map { case (jsonField, jsonDef) =>
+              jsonField -> jo(jsonField).map(JsonToTypedMap(_, jsonDef)).orElse(noneOrError(jsonField))
+            }.collect {
+              case (key, Some(value)) => key -> value
+            }
+            .filter{ case (key, _) => jo.contains(key) } //it has to be done after mapping -> first we should check missing fields.. -> noneOrError
```

Comment:
what if we do:
```  
elementType.collect {
      case (jsonField, jsonDef) if jo.contains(jsonField) => jsonField -> JsonToTypedMap(jo(jsonField).get, jsonDef))
      case (jsonField, _) if required.contains(jsonField) => throw JsonToObjectError(json, definition)
}
```
and remove noneOrError? For me it's a bit more readable?

Reply:
I've changed it. As we've spoken, right now I don't do any validation, just trimming ;)